lib/kinesis_client/stream/app_state/ecto/shard_lease.ex

```diff
@@ -20,7 +20,7 @@ defmodule KinesisClient.Stream.AppState.Ecto.ShardLease do
   def changeset(shard_lease, attrs) do
     shard_lease
     |> cast(attrs, @fields)
-    |> unique_constraint(:shard_id, name: :shard_lease_pkey)
+    |> unique_constraint([:shard_id, :app_name, :stream_name], name: :shard_lease_pkey)
```

Comment:
@tkusumo I think we need a database migration too.

Reply:
@ryanlchandler a database migration added.